aws/resource_aws_ram_resource_share.go

```diff
@@ -0,0 +1,267 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ram"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsRamResourceShare() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRamResourceShareCreate,
+		Read:   resourceAwsRamResourceShareRead,
+		Update: resourceAwsRamResourceShareUpdate,
+		Delete: resourceAwsRamResourceShareDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"allow_external_principals": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"resource_arns": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"principals": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsRamResourceShareCreate(d *schema.ResourceData, meta interface{}) error {
+	ramconn := meta.(*AWSClient).ramconn
+
+	request := &ram.CreateResourceShareInput{
+		Name:                    aws.String(d.Get("name").(string)),
+		AllowExternalPrincipals: aws.Bool(d.Get("allow_external_principals").(bool)),
+	}
+
+	if v, ok := d.GetOk("resource_arns"); ok {
+		request.ResourceArns = expandStringList(v.([]interface{}))
```

Comment:
When testing is turned on you'll probably notice immediately this should be `expandStringSet(v.(*schema.Set))` 😄 Same with `principals` below.